### PR TITLE
[aggregator] Do layout matching in change_set only sporadically

### PR DESCRIPTION
**Pull request by vusirikala**
### Description

Do layout matching in change_set only sporadically instead of every time.

### CODE REVIEW

File: aptos-move/aptos-vm-types/src/change_set.rs [hunk 1]
```diff
@@ -30,6 +32,34 @@ use std::{
     sync::Arc,
 };
 
+// Sporadically checks if the given two input type layouts match
```
[the remark this answers is not in the document]
Comment (vusirikala) [reply aptos-move/aptos-vm-types/src/change_set.rs:43]: Done.

**Review by georgemitenkov — APPROVED** [review]

File: aptos-move/aptos-vm-types/src/change_set.rs [hunk 2]
```diff
@@ -38,6 +39,34 @@ use std::{
     sync::Arc,
 };
 
+/// Sporadically checks if the given two input type layouts match.
+pub fn randomly_check_layout_matches(
```
Comment (igor-aptos) [inline aptos-move/aptos-vm-types/src/change_set.rs:43]: @vusirikala  - can you also address this TODO to use this function:

https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/block-executor/src/executor.rs#L859

you might need to move this function to a more generic file, that can be called throughout the code.

**Review by igor-aptos — APPROVED** [review]
address the linked comment before landing

File: aptos-move/aptos-vm-types/src/change_set.rs (hunk 2 above)
Comment (vusirikala) [reply aptos-move/aptos-vm-types/src/change_set.rs:40]: I tried to put the `randomly_check_layout_matches` function in value.rs where MoveTypeLayout was defined. But that crate is using an older version of rand and I got some errors due to incompatibility issues. Is it okay if I continue keeping the `randomly_check_layout_matches` in change_set.rs and use the function in executor.rs?

Comment (igor-aptos) [reply aptos-move/aptos-vm-types/src/change_set.rs:40]: if you can refer it there from here, that's fine for now

**Pull request merged**